src/astro/sql/operators/agnostic_save_file.py

```diff
@@ -96,45 +94,43 @@ def file_exists(self, output_file_path, output_conn_id=None):
 
     def convert_sql_table_to_dataframe(self):
         input_table = self.input
-        conn_type = BaseHook.get_connection(input_table.conn_id).conn_type
+        database = get_database_from_conn_id(input_table.conn_id)
 
         # Select database Hook based on `conn` type
         hook_kwargs = {
```

Comment:
@utkarsharma2 It is indeed redundant! That said, it wasn't introduced as part of this PR..!
Since this PR already grew much more than expected, would it be okay if we log a refactor ticket to review the `save_file` implementation?